### ISSUE TITLE
Change Currency Symbol to R not ZAR.

### ISSUE THREAD

**Issue by johanjvrens** [metabase/metabase#10785]
In South Africa we use the Currency Symbol R, not ZAR, we use ZAR as the currency code.

**Comment by paulrosenzweig** [comment]
It seems like the issue here is that we always pass `"en"` as the locale to `Intl.NumberFormat`.

**en**
```js
Intl.NumberFormat('en', { style: 'currency', currency: 'ZAR'}).format(123.45);
// "ZAR 123.45"
```

**en-ZA**
```js
Intl.NumberFormat('en-ZA', { style: 'currency', currency: 'ZAR'}).format(123.45);
// "R 123,45"
```

I think it'd make sense to use the current locale rather than hardcoding `"en"`.

**Comment by paoliniluis** [comment]
I believe this one is closed, so if @johanjvrens can confirm we can close this issue, thanks!

**Comment by flamber** [comment]
@paoliniluis It's still showing `ZAR` instead of `R` of Symbol on 0.38.3/`master`
![image](https://user-images.githubusercontent.com/1447303/113423636-226cc880-93cf-11eb-8ff6-5ee2db386dd7.png)

**Comment by paoliniluis** [comment]
The plot thickens:
- just a few currencies work as they should (USD, EUR and a few more), but the issue happens also with Argentine Pesos (which should display AR$, symbol, but displays ARS, code, and many many more.
- we never use the "symbol_native" field (why?)
- tried changing the locale using MetabaseSettings.get("site-locale") in the Intl.NumberFormat() function, but didn't work, as the locale is en and not en-ZA as @paulrosenzweig states above

**Comment by noahmoss** [comment]
Another example of incorrect currency i18n: we currently support using the symbol for the Vietnamese đồng, but it should come after the number rather than before it.

I think the main issue is that we're not actually using the `Intl.NumberFormat` object for currency formatting. We're using it to generate the currency symbol or name (always with the `en` locale), and then sticking it onto the number after it's been formatted with other viz settings like `separator style` and `decimals`. This approach will always cause i18n issues at the edges.

I think we need to either:
1. Let `Intl.NumberFormat` fully handle currency formatting. This might require disallowing most other number format settings for currencies. It also might return numbers in non-Latin scripts depending on the locale.
```
>> new Intl.NumberFormat("ar", {style: "currency", currency: "USD"}).format(123.12)
"١٢٣٫١٢ US$"
```
2. Handle all currency formatting in Metabase code. This would require expanding our [currency list](https://github.com/metabase/metabase/blob/a967c2a6e57109cded0a24104da75a90868aebf5/frontend/src/metabase/lib/currency.js) to include info about whether the symbol is a prefix/suffix, whether there are spaces between the symbol and the number, etc.

**Comment by bshepherdson** [comment]
I've created a new epic #27576 to capture this issue, the prefix/suffix part Noah mentioned above, and several other problems.

tl;dr is that we have the tools to do a much better job of localizing currency values, and I want to clean them up soon.
